repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -398,6 +400,11 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                         wrappedRequest.addHeader(CommonHttpHeader.TRACE_GUID.toString(),
                                 TracingHeaderHelper.createTracingHeader(traceGUID, wrappedRequest.getHeader(CommonHttpHeader.VIA.toString())));
                     }
+                    if((currentSystemModel.get().getTracingHeader() != null) &&
```

Comment:
I suppose it comes down to what a developer is more used to.

I've grown attached to and familiar with Scala's `Option`s and super don't like seeing the `null` keyword anymore. Also, it was an opportunity to inject more Java 8. 😛